src/relayserver/penalizer/PenalizerService.ts

```diff
@@ -140,15 +176,81 @@ export class PenalizerService {
       throw Error(`Failed to get transaction ${minedTransactionData.hash} from node`)
     }
     const minedTxBuffers = createWeb3Transaction(minedTx, rawTxOptions)
-    const method = this.getPenalizeRepeatedNonceMethod(minedTxBuffers, requestTx)
+    const penalizationArguments = this.getPenalizeRepeatedNonceArguments(minedTxBuffers, requestTx)
+    const method = this.getMethod(PenalizationTypes.REPEATED_NONCE, penalizationArguments)
     const isValidPenalization = await this.validatePenalization(method)
-    if (!isValidPenalization.valid) {
+    if (!validationResult.valid) {
       return {
         message: isValidPenalization.error
       }
     }
-    const penalizeTxHash = await this.executePenalization('penalizeRepeatedNonce', method)
-    return { penalizeTxHash }
+    const commitHash = this.calculateCommitHash(method)
+    const delayedPenalization: DelayedPenalization = {
+      commitHash,
+      type: PenalizationTypes.REPEATED_NONCE,
+      methodArgs: penalizationArguments
+    }
+    const commitTxHash = await this.commitAndScheduleReveal(delayedPenalization)
+    return { commitTxHash }
+  }
+
+  calculateCommitHash (method: any): PrefixedHexString {
+    const msgData = method.encodeABI()
```

Comment:
need to have a local "nonce" appended to the encoded-abi. should use the same value here, and when sending the TX

Reply:
Fixed.